plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiRecordCursors.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.hudi.common.model.HoodieLogFile;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.hadoop.realtime.HoodieRealtimeFileSplit;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.function.Function;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.trino.plugin.hudi.HudiErrorCode.HUDI_CANNOT_OPEN_SPLIT;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.hadoop.hive.serde2.ColumnProjectionUtils.READ_ALL_COLUMNS;
+import static org.apache.hadoop.hive.serde2.ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR;
+import static org.apache.hadoop.hive.serde2.ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR;
+
+class HudiRecordCursors
+{
+    private HudiRecordCursors() {}
+
+    public static RecordCursor createRealtimeRecordCursor(
+            HdfsEnvironment hdfsEnvironment,
+            ConnectorSession session,
+            Properties schema,
+            HudiSplit split,
+            List<HiveColumnHandle> dataColumns)
+    {
+        requireNonNull(session, "session is null");
+        checkArgument(dataColumns.stream().allMatch(HudiRecordCursors::isRegularColumn), "dataColumns contains non regular column");
+        HudiFile baseFile = getHudiBaseFile(split);
+        Path path = new Path(baseFile.getPath());
+
+        HdfsContext context = new HdfsContext(session);
+        Configuration configuration = hdfsEnvironment.getConfiguration(context, path);
+        return hdfsEnvironment.doAs(session.getIdentity(), () -> {
+            RecordReader<?, ?> recordReader = createRecordReader(configuration, schema, split, dataColumns);
+            @SuppressWarnings("unchecked") RecordReader<?, ? extends Writable> reader = (RecordReader<?, ? extends Writable>) recordReader;
+            return createRecordCursor(configuration, path, reader, baseFile.getLength(), schema, dataColumns);
+        });
+    }
+
+    private static RecordReader<?, ?> createRecordReader(
+            Configuration configuration,
+            Properties schema,
+            HudiSplit split,
+            List<HiveColumnHandle> dataColumns)
+    {
+        // update configuration
+        JobConf jobConf = new JobConf(configuration);
+        jobConf.setBoolean(READ_ALL_COLUMNS, false);
+        jobConf.set(READ_COLUMN_IDS_CONF_STR, join(dataColumns, HiveColumnHandle::getBaseHiveColumnIndex));
+        jobConf.set(READ_COLUMN_NAMES_CONF_STR, join(dataColumns, HiveColumnHandle::getBaseColumnName));
+        schema.stringPropertyNames()
+                .forEach(name -> jobConf.set(name, schema.getProperty(name)));
+        refineCompressionCodecs(jobConf);
```

Comment:
Just for my understanding, what's the intention of setting compression codecs here?